pyccel/codegen/printing/ccode.py

```diff
@@ -1088,6 +1088,13 @@ def _print_AliasAssign(self, expr):
 
         lhs = self._print(lhs.name)
         rhs = self._print(rhs)
+
+        # the below condition handles the case of reassinging a pointer to an array view.
+        # setting the pointer's is_view attribute to false so it can be ignored by the free_pointer function.
+        if isinstance(expr.lhs, Variable) and expr.lhs.is_ndarray \
+                and isinstance(expr.rhs, Variable) and expr.lhs.is_ndarray and expr.rhs.is_pointer:
+            return '{0} = {1};\n{0}.is_view = false;'.format(lhs, rhs)
+
```

Comment:
can you please provide an example and where it fails?

Reply:
``` python
a = np.array([1, 2, 3, 4, 5])
b = a[1:]
e = b # this change is to fix this problem case because we are allowing reassinging pointer freely
```

``` C
int main()
{
    t_ndarray a;
    t_ndarray b;
    t_ndarray e;



    a = array_create(1, (int32_t[]){5}, nd_int64);
    int64_t array_dummy_0001[] = {1, 2, 3, 4, 5};
    memcpy(a.nd_int64, array_dummy_0001, a.buffer_size);

    b = array_slicing(a, new_slice(1, 5, 1));
    e = b;
    e.is_view = false;  //this change is to add this line so I can ignore the variable e in the garbage collecting
                                // because it copies the information of b and b is a view
    free_array(a);
    free_pointer(b);
    free_pointer(e);
    return 0;
}
```

Reply:
Do you mean that `e` `is_ndarray` instead of `is_pointer`? That's not right :/ 

It looks to me like the actual problem is here:
https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L1308
`rhs.is_allocatable` should be `(rhs.is_ndarray or rhs.is_pointer)`

But I thought the pointer case was handled [here](https://github.com/pyccel/pyccel/blob/master/pyccel/parser/semantic.py#L632)

Reply:
the problem is not from pyccel side.
the problem is we can't know If the variable in C is a view or not statically so i need to add somethings to make a difference so the garbage collector can  ignore the pointers that don't own (shape/strides)

Reply:
So this tells C that `b` owns its shape/strides, but not the array data, but `e` owns none of it's pointers?

Reply:
exactly